# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/mepo/0003-zig-0.13-std.ChildProcess-std.process.Child.patch
# Copyright (C) 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

From 65113031c6b0ce63e46f8fba43df268a32a1bb65 Mon Sep 17 00:00:00 2001
From: "lauren n. liberda" <lauren@selfisekai.rocks>
Date: Sat, 8 Jun 2024 02:40:54 +0200
Subject: [PATCH] zig 0.13: std.ChildProcess -> std.process.Child

---
 src/api/shellpipe_async.zig | 2 +-
 src/api/shellpipe_sync.zig  | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/api/shellpipe_async.zig b/src/api/shellpipe_async.zig
index 9ebcfbf..e0f722e 100644
--- a/src/api/shellpipe_async.zig
+++ b/src/api/shellpipe_async.zig
@@ -60,7 +60,7 @@ fn async_shellpipe_run_catch_errors(mepo: *Mepo, unique_handle_id: i8, cmd: []co
     const max_output_bytes: usize = 50 * 1024;
     var arena = std.heap.ArenaAllocator.init(mepo.allocator);
     defer arena.deinit();
-    var child = std.ChildProcess.init(argv[0..], arena.allocator());
+    var child = std.process.Child.init(argv[0..], arena.allocator());
     child.stdin_behavior = .Ignore;
     child.stdout_behavior = .Pipe;
     const env = try get_env_vars(mepo, mepo.allocator);
diff --git a/src/api/shellpipe_sync.zig b/src/api/shellpipe_sync.zig
index 133b8c7..d0f6186 100644
--- a/src/api/shellpipe_sync.zig
+++ b/src/api/shellpipe_sync.zig
@@ -32,7 +32,7 @@ fn shellpipe_sync(mepo: *Mepo, cmd: []const u8) !void {
     try mepo.blit();
     const env_vars = try get_env_vars(mepo, arena.allocator());
     const args = [_][]const u8{ "sh", "-c", cmd };
-    const process_result = try std.ChildProcess.run(.{
+    const process_result = try std.process.Child.run(.{
         .allocator = arena.allocator(),
         .argv = args[0..],
         .env_map = &env_vars,
